MessageTTL checking fails with deserialization errors

**Describe the bug**

MessageTTL checking fails with deserialization errors.

```
java.lang.RuntimeException: Could not deserialize object [MessageRecord]. Deserialization stuck at offset 24 of length 69
    at io.camunda.zeebe.msgpack.UnpackedObject.wrap(UnpackedObject.java:38)
    at io.camunda.zeebe.engine.api.records.RecordBatchEntry.createEntry(RecordBatchEntry.java:73)
    at io.camunda.zeebe.engine.api.records.RecordBatch.appendRecord(RecordBatch.java:48)
    at io.camunda.zeebe.streamprocessor.BufferedTaskResultBuilder.appendCommandRecord(BufferedTaskResultBuilder.java:47)
    at io.camunda.zeebe.engine.processing.message.MessageTimeToLiveChecker.lambda$execute$0(MessageTimeToLiveChecker.java:90)
```

This will fail the processing actor and prevent processing on this partition. When the (experimental) feature flag `enableMessageTtlCheckerAsync` is used, a different actor fails and only prevents further MessageTTL checking but not processing.

This is a regression, introduced in https://github.com/camunda/zeebe/commit/e1a6cae69c17325fc71a8ee92022a70d969bd0da
It affects versions >= 8.1.9 and >= 8.2.0.


**To Reproduce**

Run a broker that is leader for multiple partitions. When messages from two different partitions are expired at the same time, there is unsafe concurrent access to the writer of a shared record value. 

**Expected behavior**

<!-- A clear and concise description of what you expected to happen. -->

**Log/Stacktrace**

<!-- If possible add the full stacktrace or Zeebe log which contains the issue. -->

<details><summary>Full Stacktrace</summary>
 <p>

```
java.lang.RuntimeException: Could not deserialize object [MessageRecord]. Deserialization stuck at offset 24 of length 69
    at io.camunda.zeebe.msgpack.UnpackedObject.wrap(UnpackedObject.java:38)
    at io.camunda.zeebe.engine.api.records.RecordBatchEntry.createEntry(RecordBatchEntry.java:73)
    at io.camunda.zeebe.engine.api.records.RecordBatch.appendRecord(RecordBatch.java:48)
    at io.camunda.zeebe.streamprocessor.BufferedTaskResultBuilder.appendCommandRecord(BufferedTaskResultBuilder.java:47)
    at io.camunda.zeebe.engine.processing.message.MessageTimeToLiveChecker.lambda$execute$0(MessageTimeToLiveChecker.java:90)
    at io.camunda.zeebe.engine.state.message.DbMessageState.lambda$visitMessagesWithDeadlineBeforeTimestamp$2(DbMessageState.java:358)
    at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.visit(TransactionalColumnFamily.java:390)
    at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.lambda$forEachInPrefix$19(TransactionalColumnFamily.java:369)
    at io.camunda.zeebe.db.impl.rocksdb.transaction.ColumnFamilyContext.withPrefixKey(ColumnFamilyContext.java:112)
    at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.forEachInPrefix(TransactionalColumnFamily.java:353)
    at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.lambda$whileTrue$8(TransactionalColumnFamily.java:174)
    at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.lambda$ensureInOpenTransaction$18(TransactionalColumnFamily.java:308)
    at io.camunda.zeebe.db.impl.rocksdb.transaction.DefaultTransactionContext.runInNewTransaction(DefaultTransactionContext.java:61)
    at io.camunda.zeebe.db.impl.rocksdb.transaction.DefaultTransactionContext.runInTransaction(DefaultTransactionContext.java:33)
    at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.ensureInOpenTransaction(TransactionalColumnFamily.java:307)
    at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.whileTrue(TransactionalColumnFamily.java:174)
    at io.camunda.zeebe.engine.state.message.DbMessageState.visitMessagesWithDeadlineBeforeTimestamp(DbMessageState.java:351)
    at io.camunda.zeebe.engine.processing.message.MessageTimeToLiveChecker.execute(MessageTimeToLiveChecker.java:76)
    at io.camunda.zeebe.streamprocessor.ProcessingScheduleServiceImpl.lambda$toRunnable$6(ProcessingScheduleServiceImpl.java:137)
    at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:94)
    at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:45)
    at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:119)
    at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:106)
    at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:87)
    at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:198)
Caused by: io.camunda.zeebe.msgpack.spec.MsgpackReaderException: Unable to determine string type, found unknown header byte 0x00 at reader offset 23
    at io.camunda.zeebe.msgpack.spec.MsgPackReader.exceptionOnUnknownHeader(MsgPackReader.java:474)
    at io.camunda.zeebe.msgpack.spec.MsgPackReader.readStringLength(MsgPackReader.java:140)
    at io.camunda.zeebe.msgpack.value.StringValue.read(StringValue.java:96)
    at io.camunda.zeebe.msgpack.value.ObjectValue.read(ObjectValue.java:91)
    at io.camunda.zeebe.msgpack.UnpackedObject.wrap(UnpackedObject.java:32)
    ... 24 more
```

</p>
</details>

**Environment:**
- OS: <!-- [e.g. Linux] -->
- Zeebe Version: <!-- [e.g. 0.20.0] -->
- Configuration: <!-- [e.g. exporters etc.] -->

--- a/engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessageTimeToLiveChecker.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessageTimeToLiveChecker.java
@@ -33,7 +33,7 @@ import org.agrona.collections.MutableInteger;
  */
 public final class MessageTimeToLiveChecker implements Task {
 
-  private static final MessageRecord EMPTY_DELETE_MESSAGE_COMMAND =
+  private final MessageRecord emptyDeleteMessageCommand =
       new MessageRecord().setName("").setCorrelationKey("").setTimeToLive(-1L);
 
   /** This determines the duration that the TTL checker is idle after it completes an execution. */
@@ -88,7 +88,7 @@ public final class MessageTimeToLiveChecker implements Task {
 
               final boolean stillFitsInResult =
                   taskResultBuilder.appendCommandRecord(
-                      expiredMessageKey, MessageIntent.EXPIRE, EMPTY_DELETE_MESSAGE_COMMAND);
+                      expiredMessageKey, MessageIntent.EXPIRE, emptyDeleteMessageCommand);
               return stillFitsInResult && counter.incrementAndGet() < batchLimit;
             });
 

